### DISCUSSION TITLE
Convert my site from RsT to Markdown

### DISCUSSION THREAD

**Discussion by giampaolo** [getpelican/pelican#3408] [General]
Hi! Are there tools or guides on how to do this in a somewhat automated fashion?
I have found this gist https://gist.github.com/zaiste/77a946bbba73f5c4d33f3106a494e6cd which does the conversion, but Pelican is not fully happy with it (e.g. it complains about missing `title` and other metadata). Also, I'm not sure how to explicitly tell Pelican to start looking for .md files rather than .rst (via conf?). 

Alternatively, I wonder if perhaps I can start using .md only for new blog posts, and keep the old ones as .rst, so that I can avoid the effort of converting the entire site.

Thank you for this great tool!

**Comment by justinmayer** [comment]
Ciao @giampaolo. Thank you for the kind words. You can indeed use both Markdown and reStructuredText files in the same Pelican project at the same time. As long as you have installed Pelican along with the Python-Markdown library — for example, via `python -m pip install pelican[markdown]` — then Pelican should automatically detect and process Markdown files it finds in your source content.

Regarding how to convert reStructuredText files to Markdown, I am not aware of any existing guides, so I decided to create my own guide: [Convert reStructuredText to Markdown](https://hackercodex.com/guide/convert-restructuredtext-to-markdown/)

Would you please have a look and let me know your thoughts? I would be curious to know whether it is useful for you 😊

**Reply by giampaolo** [reply]
Wow! I definitively didn't expect a full guide! I will try this out and let you know.

**Reply by justinmayer** [reply]
Did my guide shed any light on this topic? Helpful at all? Any suggestions for improvement?

**Reply by giampaolo** [reply]
Hello Justin. Not quite. It was difficult to install `fd` and `sd` on Linux, so I replaced those commands with `find` and `sed` (but that was the easy part). The hard part is that pandoc converts links like this:

```
`sendfile`

-* `sendfile <https://sendfile.com/>`__ 
```
Into:

```
[sendfile]{.title-ref}
```

Not sure how to handle that, I'm still investigating.

**Reply by justinmayer** [reply]
I suspect you may have to use `find`/`sed` to search and replace those problematic reST reference links *before* you run them through Pandoc.

**Comment by mcepl** [comment]
I know it is too late, but for whoever finds this via Internet Search, `rst2myst` provides much better conversion from rST to Markdown than `pandoc`.
